solutions/nx-monorepo/apps/app/next.config.js

```diff
@@ -0,0 +1,17 @@
+//@ts-check
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { withNx } = require('@nrwl/next/plugins/with-nx');
+
+/**
+ * @type {import('@nrwl/next/plugins/with-nx').WithNxOptions}
+ **/
+const nextConfig = {
+  nx: {
+    // Set this to true if you would like to to use SVGR
+    // See: https://github.com/gregberge/svgr
+    svgr: false,
+  },
+};
+
+module.exports = withNx(nextConfig);
```

Comment:
What does `@nrwl/next/plugins/with-nx` do? I couldn't find any documentation about it, I'm a bit scared this can break with the next major of Next.

Reply:
Here's the source: https://github.com/nrwl/nx/blob/master/packages/next/plugins/with-nx.ts - it does some webpack shenanigans for environment variable detection to ensure proper cache detection, monorepo, and workspace setup.|

Here's some loose docs as well: https://nx.dev/packages/next